eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1778,22 +1789,13 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
   }
 
   onTransition {
-    case _ -> CLOSING =>
-      PendingRelayDb.getPendingFailsAndFulfills(nodeParams.db.pendingRelay, nextStateData.asInstanceOf[HasCommitments].channelId) match {
-        case Nil =>
-          log.debug("nothing to replay")
-        case cmds =>
-          log.info("replaying {} unacked fulfills/fails", cmds.size)
-          cmds.foreach(self ! _) // they all have commit = false
-      }
-    case SYNCING -> (NORMAL | SHUTDOWN) =>
-      PendingRelayDb.getPendingFailsAndFulfills(nodeParams.db.pendingRelay, nextStateData.asInstanceOf[HasCommitments].channelId) match {
-        case Nil =>
-          log.debug("nothing to replay")
-        case cmds =>
-          log.info("replaying {} unacked fulfills/fails", cmds.size)
-          cmds.foreach(self ! _) // they all have commit = false
-          self ! CMD_SIGN() // so we can sign all of them at once
+    case state -> nextState =>
+      (state, nextState, nextStateData) match {
+        case (_, CLOSING, data: HasCommitments) if data.commitments.hasPendingOrProposedHtlcs =>
```

Comment:
On the contrary, we want a DB call here in all cases for safety.
It allows us to detect if there's a bug that creates incoherences between channel state and pendingRelayDb.
And the previous code was clearer by only focusing on the states we're transitioning to/from.